src/util/floatingpoint.cpp

```diff
@@ -614,47 +629,79 @@ static FloatingPointLiteral constructorHelperBitVector(
       FloatingPointLiteral rounded(
-          symfpu::convertFloatToFloat(exactFormat, ct, rm, exactFloat));
+          symfpu::convertFloatToFloat(exactFormat, size, rm, exactFloat));
       return rounded;
 #else
       Unreachable() << "no concrete implementation of FloatingPointLiteral";
 #endif
     }
     Unreachable() << "Constructor helper broken";
-  }
-  
-  FloatingPoint::FloatingPoint (const FloatingPointSize &ct, const RoundingMode &rm, const Rational &r) :
-    fpl(constructorHelperRational(ct, rm, r)),
-    t(ct) {}
+}
 
-  
-  FloatingPoint FloatingPoint::makeNaN (const FloatingPointSize &t) {
+FloatingPoint::FloatingPoint(const FloatingPointSize& size,
+                             const RoundingMode& rm,
+                             const Rational& r)
+    : fpl(constructorHelperRational(size, rm, r)), t(size)
+{
+}
+
+FloatingPoint FloatingPoint::makeNaN(const FloatingPointSize& t)
+{
 #ifdef CVC4_USE_SYMFPU
-    return FloatingPoint(
-        t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeNaN(t));
+  return FloatingPoint(
+      t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeNaN(t));
 #else
-    return FloatingPoint(2, 2, BitVector(4U,0U));
+  return FloatingPoint(2, 2, BitVector(4U, 0U));
 #endif
-  }
+}
 
-  FloatingPoint FloatingPoint::makeInf (const FloatingPointSize &t, bool sign) {
+FloatingPoint FloatingPoint::makeInf(const FloatingPointSize& t, bool sign)
+{
 #ifdef CVC4_USE_SYMFPU
-    return FloatingPoint(
-        t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeInf(t, sign));
+  return FloatingPoint(
+      t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeInf(t, sign));
 #else
-    return FloatingPoint(2, 2, BitVector(4U,0U));
+  return FloatingPoint(2, 2, BitVector(4U, 0U));
 #endif
-  }
+}
 
-  FloatingPoint FloatingPoint::makeZero (const FloatingPointSize &t, bool sign) {
+FloatingPoint FloatingPoint::makeZero(const FloatingPointSize& t, bool sign)
+{
 #ifdef CVC4_USE_SYMFPU
-    return FloatingPoint(
-        t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeZero(t, sign));
+  return FloatingPoint(
+      t, symfpu::unpackedFloat<symfpuLiteral::traits>::makeZero(t, sign));
 #else
-    return FloatingPoint(2, 2, BitVector(4U,0U));
+  return FloatingPoint(2, 2, BitVector(4U, 0U));
 #endif
+}
+
+FloatingPoint FloatingPoint::makeMinSubnormal(const FloatingPointSize& t,
+                                              bool sign)
+{
+  if (sign)
+  {
+    return FloatingPoint(t,
+                         BitVector::mkMinSigned(t.exponent() + 1)
+                             .concat(BitVector(t.significand() - 1, 1u)));
   }
+  return FloatingPoint(
+      t,
+      BitVector(t.exponent() + 1).concat(BitVector(t.significand() - 1, 1u)));
+}
 
+FloatingPoint FloatingPoint::makeMaxSubnormal(const FloatingPointSize& t,
+                                              bool sign)
+{
+  if (sign)
+  {
+    return FloatingPoint(t,
+                         BitVector::mkMinSigned(t.exponent() + 1)
```

Comment:
See similar to above.  The utility functions in `FloatingPointSize` are there to help you not make mistakes about packed vs. unpacked, etc.